ocaml/xapi/xapi_xenops.ml

```diff
@@ -667,10 +667,19 @@ module MD = struct
                   |> List.assoc Xapi_globs.vgpu_type_id in
     let uuid = vgpu.Db_actions.vGPU_uuid in
     let extra_args = vgpu.Db_actions.vGPU_extra_args in
+    let config_file = match Xapi_vgpu_type.Vendor_nvidia.is_host_driver_support_multi_vgpu () with
+      | true -> None
+      | false -> (* Host driver does not support multiple vGPU, prepare config_file for the legency mode *)
+        let config_dir =  "/usr/share/nvidia/vgx" in
+        let model_name_in_config_file = Db.VGPU_type.get_model_name ~__context ~self:vgpu_type (* like "GRID M10-8Q" *)
+                                        |> String.split_on_char ' ' |> String.concat "_" |> String.lowercase_ascii in
+        let path = Printf.sprintf "%s/%s.conf" config_dir model_name_in_config_file in
```

Comment:
Is this guaranteed to work for all models? A safer alternative is to repurpose `Nvidia_compat.create_compat_lookup_file` in xapi_vgpu_type.ml to add the `config_file` path into the `internal_config` of the types, so that we can just get it here like the `type_id` above. We could add the path to `internal_config` only when then driver check fails, so that we only have to do that check once.

Reply:
how to  identify which `config_file` belong to which type, it may still need to use the method above.
It is a good point to store it in xapi db and only compute once is a good point

Reply:
That is already handled in `Nvidia_compat`. You just need to change the bit in `create_compat_lookup_file` that writes to the file, to instead updated the DB.

Reply:
okay, I will check it